exp/addons/internal/controllers/clusterresourcesetbinding_controller.go

```diff
@@ -77,8 +78,19 @@ func (r *ClusterResourceSetBindingReconciler) Reconcile(ctx context.Context, req
 		// Error reading the object - requeue the request.
 		return ctrl.Result{}, err
 	}
-
-	cluster, err := util.GetOwnerCluster(ctx, r.Client, binding.ObjectMeta)
+	var clusterName = binding.Spec.ClusterName
+	if clusterName == "" {
+		// Update the clusterName field of the existing ClusterResourceSetBindings with ownerReferences.
+		// More details please refer to: https://github.com/kubernetes-sigs/cluster-api/issues/7669.
+		if clusterName, err = getClusterNameFromOwnerRef(binding.ObjectMeta); clusterName != "" && err == nil {
```

Comment:
Once inside here let's:
1) Return an error if getCluserNameFromOwnerRef returns an error
2) Return a requeue if there's no error.

Reply:
> Return a requeue if there's no error.

Sorry, I didn't understand. 

I will update the `binding`. If the update is successful, it will automatically enter the Reconcile next time.